course4/homework-string/string.cc

```diff
@@ -4,6 +4,129 @@ namespace course {
 
 string::string()
 {
+	mStr = std::make_unique<char[]>(1);
```

Comment:
While correct, this is unpleasant to the user. Well-behaving classes should not do anything expensive in the default constructor.